src/solver/prelu/backward_prelu.cpp

```diff
@@ -0,0 +1,327 @@
+            {"VIEW_DIMS", VIEW_DIMS},
+            {"INPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype},
+            {"OUTPUT_TYPE", output_dtype == "bfloat16" ? "ushort" : output_dtype},
+        };
+        result.construction_params.push_back(make_hip_kernel(
+            {LOCAL_SIZE_SW_BWD}, {size}, "MIOpenPReLU.cpp", "PReLUMWBackward", build_params));
+    }
+
+    /* Phase 2: Reduce gradient. */
+    {
+        auto build_params = KernelBuildParameters{
+            {"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
+            {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
+            {"MIOPEN_USE_FP64", static_cast<int>(dtype == miopenDouble)},
+            {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
+            {"OUTPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype},
+            {"REDUCE_SIZE", LOCAL_SIZE_MW_REDUCE_BWD},
+        };
+        result.construction_params.push_back(
+            make_hip_kernel({LOCAL_SIZE_MW_REDUCE_BWD},
+                            {problem.GetdWeightDesc().GetElementSize() * LOCAL_SIZE_MW_REDUCE_BWD},
+                            "MIOpenReduceSum.cpp",
+                            "Reduce1dSum",
+                            build_params));
+    }
+
+    result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+            auto params = raw_params.CastTo<miopen::prelu::InvokeParams>();
+
+            auto elapsed = 0.f;
+            HipEventPtr start;
+            HipEventPtr stop;
+
+            bool reset_profiling_state = false;
+            if(handle_.IsProfilingEnabled())
+            {
+                reset_profiling_state = true;
+                handle_.EnableProfiling(false);
+                hipStreamSynchronize(handle_.GetStream());
+                start = miopen::make_hip_event();
+                stop  = miopen::make_hip_event();
+                hipEventRecord(start.get(), handle_.GetStream());
+            }
+
+            int kernelCnt = 0;
+
+            /* Phase 1: Calc gradient for each elements. */
+            {
+                auto input_tv       = get_inner_expanded_tv<VIEW_DIMS>(deref(params.inputDesc));
+                auto output_grad_tv = get_inner_expanded_tv<VIEW_DIMS>(deref(params.doutputDesc));
+                auto input_grad_tv  = get_inner_expanded_tv<VIEW_DIMS>(deref(params.dinputDesc));
+                auto kernel         = handle_.Run(kernels[kernelCnt++]);
+                kernel(params.input,
+                       params.weight,
+                       params.doutput,
+                       params.dinput,
+                       params.workspace,
+                       (ulong)deref(params.inputDesc).GetElementSize(),
+                       input_tv,
+                       output_grad_tv,
+                       input_grad_tv);
+            }
+
+            /* Phase 2: Reduce gradient. */
```

Comment:
I think the Reduce calculation method has been solved with a single kernel. Is it possible to check the results compared to moDNN performance?

Reply:
I think that you expect `SumFwdContiguous` kernel in `MIOpenReduceSum.cpp` to be used for handling Reduce calcuation. Well, this kernel must be called twice as PReLU reduce all dimension except the second dim (one for the first dim, one for all dims after the 2nd). Calling twice will cause lauch kernel overhead, so I made a new kernel that Reduce from N dims to 1 dim. I have benchmarked and uploaded the results to this file [(P,R)ReLU.xlsx](https://github.com/user-attachments/files/15815166/P.R.ReLU.xlsx), there are results of both Modnn and MIOpen.